lib/OpenQA/Worker.pm

```diff
@@ -287,9 +287,27 @@ sub init {
             Mojo::IOLoop->stop();
         });
 
-    # initialize clients to connect to the web UIs
+
     my $global_settings              = $settings->global_settings;
     my $webui_host_specific_settings = $settings->webui_host_specific_settings;
+
+    # Store the packages list of the environment the worker runs in
+    if (my $cmd = $global_settings->{PACKAGES_CMD}) {
+        my $pool_directory = $self->pool_directory;
+        log_info("Gathering package information: $cmd");
+
+        my $packages_filename = $pool_directory . '/worker_packages.txt';
+        if (system("$cmd > $packages_filename") != 0) {
+            log_error('The PACKAGES_CMD command could not be executed');
+            return 1;
+        }
+        if (-s $packages_filename == 0) {
```

Comment:
This will result in the warning `Use of uninitialized value in numeric eq (==)` if the file is not present at all.

```suggestion
        unless (-s $packages_filename) {
```